tests/unit/test_application.py

```diff
@@ -0,0 +1,45 @@
+from unittest import TestCase, mock
+from vorta.application import VortaApp
+
```

Comment:
@m3nu @real-yfprojects I created unit tests for the application file testing the `check_darwin_permissions()`. Running these unit tests individually work, however when I run them with `pytest` they fail. Do you have any ideas on how to fix this? I think it is due to the decorator `fixture` that is being used for every test.

```
@pytest.fixture(scope='function', autouse=True)
def init_db(qapp, qtbot, tmpdir_factory):
```

This is the full output: 
```
tests/network_manager/test_network_manager.py::test_decode_ssid[ssid_bytes1-\U0001f596] PASSED
tests/network_manager/test_network_manager.py::test_decode_ssid[ssid_bytes2-\\x00\\x01\\x02\\n"'] PASSED
tests/unit/test_application.py::TestApplication::test_check_darwin_permissions_with_full_access_disabled FAILED
tests/unit/test_application.py::TestApplication::test_check_darwin_permissions_with_full_access_enabled PASSED
tests/unit/test_application.py::TestApplication::test_check_darwin_permissions_with_full_access_enabled ERROR

=========================================================================================================================================== ERRORS ===========================================================================================================================================
________________________________________________________________________________________________ ERROR at teardown of TestApplication.test_check_darwin_permissions_with_full_access_enabled _________________________________________________________________________________________________

qapp = <vorta.application.VortaApp object at 0x1158d85e0>, qtbot = <pytestqt.qtbot.QtBot object at 0x130c09040>
tmpdir_factory = TempdirFactory(_tmppath_factory=TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0x115914220>, _basetemp=PosixPath('/private/var/folders/_c/4jv_h65x1zb0mzfr4n29p71r0000gn/T/pytest-of-ted/pytest-96')))

    @pytest.fixture(scope='function', autouse=True)
    def init_db(qapp, qtbot, tmpdir_factory):
        tmp_db = tmpdir_factory.mktemp('Vorta').join('settings.sqlite')
        mock_db = SqliteDatabase(
            str(tmp_db),
            pragmas={
                'journal_mode': 'wal',
            },
        )
        vorta.store.connection.init_db(mock_db)
    
        default_profile = BackupProfileModel(name='Default')
        default_profile.save()
    
        new_repo = RepoModel(url='i0fi93@i593.repo.borgbase.com:repo')
        new_repo.encryption = 'none'
        new_repo.save()
    
        default_profile.repo = new_repo.id
        default_profile.dont_run_on_metered_networks = False
        default_profile.validation_on = False
        default_profile.save()
    
        test_archive = ArchiveModel(snapshot_id='99999', name='test-archive', time=dt(2000, 1, 1, 0, 0), repo=1)
        test_archive.save()
    
        test_archive1 = ArchiveModel(snapshot_id='99998', name='test-archive1', time=dt(2000, 1, 1, 0, 0), repo=1)
        test_archive1.save()
    
        source_dir = SourceFileModel(dir='/tmp/another', repo=new_repo, dir_size=100, dir_files_count=18, path_isdir=True)
        source_dir.save()
    
        qapp.main_window.deleteLater()
        del qapp.main_window
        qapp.main_window = MainWindow(qapp)  # Re-open main window to apply mock data in UI
    
        yield
    
        qapp.jobs_manager.cancel_all_jobs()
        qapp.backup_finished_event.disconnect()
>       qapp.scheduler.schedule_changed.disconnect()
E       TypeError: disconnect() failed between 'schedule_changed' and all its connections

tests/conftest.py:101: TypeError
------------------------------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------------------------------
WARNING  vorta.scheduler:scheduler.py:72 Failed to connect to DBUS interface to detect sleep/resume events
========================================================================================================================================== FAILURES ==========================================================================================================================================
__________________________________________________________________________________________________________ TestApplication.test_check_darwin_permissions_with_full_access_disabled ___________________________________________________________________________________________________________

__wrapped_mock_method__ = <function NonCallableMock.assert_called_with at 0x10c2f1f70>, args = (<MagicMock name='logger.info' id='5112378128'>, 'Skipping check due to setting'), kwargs = {}, __tracebackhide__ = True
msg = "expected call not found.\nExpected: info('Skipping check due to setting')\nActual: not called.", __mock_self = <MagicMock name='logger.info' id='5112378128'>

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
>           __wrapped_mock_method__(*args, **kwargs)

../../vorta/env/lib/python3.9/site-packages/pytest_mock/plugin.py:444: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='logger.info' id='5112378128'>, args = ('Skipping check due to setting',), kwargs = {}, expected = "info('Skipping check due to setting')", actual = 'not called.'
error_message = "expected call not found.\nExpected: info('Skipping check due to setting')\nActual: not called."

    def assert_called_with(self, /, *args, **kwargs):
        """assert that the last call was made with the specified arguments.
    
        Raises an AssertionError if the args and keyword args passed in are
        different to the last call to the mock."""
        if self.call_args is None:
            expected = self._format_mock_call_signature(args, kwargs)
            actual = 'not called.'
            error_message = ('expected call not found.\nExpected: %s\nActual: %s'
                    % (expected, actual))
>           raise AssertionError(error_message)
E           AssertionError: expected call not found.
E           Expected: info('Skipping check due to setting')
E           Actual: not called.

/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/mock.py:898: AssertionError

During handling of the above exception, another exception occurred:

self = <tests.unit.test_application.TestApplication testMethod=test_check_darwin_permissions_with_full_access_disabled>, mock_logger = <MagicMock name='logger' id='5106011776'>, mock_settings = <MagicMock name='get' id='5117555184'>

    @mock.patch('vorta.store.models.SettingsModel.get')
    @mock.patch('vorta.application.logger')
    def test_check_darwin_permissions_with_full_access_disabled(self, mock_logger, mock_settings):
        mock_settings.return_value.value = False
    
        self.app.check_darwin_permissions()
    
        # Assert that the correct methods were called on the mock objects
        mock_settings.assert_called_with(key='check_full_disk_access')
>       mock_logger.info.assert_called_with('Skipping check due to setting')
E       AssertionError: expected call not found.
E       Expected: info('Skipping check due to setting')
E       Actual: not called.

tests/unit/test_application.py:42: AssertionError
------------------------------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------------------------------
WARNING  vorta.scheduler:scheduler.py:72 Failed to connect to DBUS interface to detect sleep/resume events
================================================================================================================================== short test summary info ===================================================================================================================================
FAILED tests/unit/test_application.py::TestApplication::test_check_darwin_permissions_with_full_access_disabled - AssertionError: expected call not found.
ERROR tests/unit/test_application.py::TestApplication::test_check_darwin_permissions_with_full_access_enabled - TypeError: disconnect() failed between 'schedule_changed' and all its connections
===================================================================================================================== 1 failed, 123 passed, 1 skipped, 1 error in 36.49s =====================================================================================================================
```

Reply:
You are using the unittest testing suite. I dont whether that is the reason for failing tests but new tests should use the pytest way of writing tests.

Reply:
Ok, thank you! I am writing a new test using pytest. Since we don't want to interact with the prompt, for now I am starting with a simple test that checks and unchecks the setting to ensure it doesn't crash.

Reply:
The two tests you already wrote, can also be done with pytest.